src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinActivityConversionExtensions.cs

```diff
@@ -184,86 +100,251 @@ private static string EncodeTraceId(ActivityTraceId traceId, bool useShortTraceI
         };
     }
 
-    internal struct TagEnumerationState : PeerServiceResolver.IPeerServiceState
+    private static void ExtractActivityTags(Activity activity, ref PooledList<KeyValuePair<string, object?>> tags)
     {
-        public PooledList<KeyValuePair<string, object?>> Tags;
-
-        public string? PeerService { get; set; }
-
-        public int? PeerServicePriority { get; set; }
-
-        public string? HostName { get; set; }
-
-        public string? IpAddress { get; set; }
+        foreach (ref readonly var tag in activity.EnumerateTagObjects())
+        {
+            if (tag.Key != ZipkinErrorFlagTagName && tag.Key != SpanAttributeConstants.StatusCodeKey)
+            {
+                PooledList<KeyValuePair<string, object?>>.Add(ref tags, tag);
+            }
 
-        public long Port { get; set; }
+            cachedTags[tag.Key] = tag.Value;
+        }
+    }
 
-        public StatusCode? StatusCode { get; set; }
+    private static void ExtractActivityStatus(Activity activity, ref PooledList<KeyValuePair<string, object?>> tags)
+    {
+        // When status is set on Activity using the native Status field in activity,
+        // which was first introduced in System.Diagnostic.DiagnosticSource 6.0.0.
+        if (activity.Status != ActivityStatusCode.Unset)
+        {
+            if (activity.Status == ActivityStatusCode.Ok)
+            {
+                PooledList<KeyValuePair<string, object?>>.Add(
+                    ref tags,
+                    new KeyValuePair<string, object?>(
+                        SpanAttributeConstants.StatusCodeKey,
+                        "OK"));
+            }
 
-        public string StatusDescription { get; set; }
+            // activity.Status is Error
+            else
+            {
+                string statusDescription =
+                    activity.StatusDescription
```

Comment:
Existing code logic set this to `activity.StatusDescription ?? string.Empty` in this code branch (`if (activity.Status != ActivityStatusCode.Unset)`). But I'm not sure if this should change, though.